app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/model/Exam.kt

```diff
@@ -15,34 +16,37 @@ import org.simpleframework.xml.Root
  * Note: This model is based on the TUMOnline web service response format for a
  * corresponding request.
  */
-@Root(name = "row", strict = false)
-data class Exam(@field:Element(name = "lv_titel")
-                var course: String = "",
-                @field:Element(name = "lv_credits", required = false)
-                var credits: String = "0",
-                @field:Element(name = "datum", required = false)
-                var date: DateTime = DateTime(),
-                @field:Element(name = "pruefer_nachname", required = false)
-                var examiner: String = "",
-                @field:Element(name = "uninotenamekurz", required = false)
-                var grade: String = "",
-                @field:Element(name = "modus", required = false)
-                var modus: String = "",
-                @field:Element(name = "studienidentifikator")
-                var programID: String = "",
-                @field:Element(name = "lv_semester", required = false)
-                var semester: String = "") : Comparable<Exam>, SimpleStickyListHeadersAdapter.SimpleStickyListItem {
+@Xml(name = "row")
+data class Exam(
+        @PropertyElement(name = "lv_titel")
+        val course: String,
+        @PropertyElement(name = "lv_credits")
+        val credits: String? = null,
+        @PropertyElement(name = "datum", converter = DateConverter::class)
+        val date: DateTime? = null,
+        @PropertyElement(name = "pruefer_nachname")
+        val examiner: String? = null,
+        @PropertyElement(name = "uninotenamekurz")
+        val grade: String? = null,
+        @PropertyElement(name = "modus")
+        val modus: String? = null,
+        @PropertyElement(name = "studienidentifikator")
+        val programID: String,
+        @PropertyElement(name = "lv_semester")
+        val semester: String = ""
+) : Comparable<Exam>, SimpleStickyListHeadersAdapter.SimpleStickyListItem {
 
     override fun getHeadName() = semester
 
     override fun getHeaderId() = semester
 
-    override fun compareTo(other: Exam): Int =
-            if (headerId == other.headerId) {
-                course.compareTo(other.course) * (-1)
-            } else {
-                headerId.compareTo(other.headerId) * (-1)
-            }
+    override fun compareTo(other: Exam): Int {
+        return if (this.semester == other.semester) {
+            course.compareTo(other.course) * (-1)
+        } else {
+            semester.compareTo(other.semester)
+        }
```

Comment:
This should be [`compareBy()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.comparisons/compare-by.html)

Reply:
Why?

Reply:
I changed the sorting, but decided to use `compareByDescending()` and chain additional `thenBy()` calls.